sql/structured.go

```diff
@@ -51,6 +52,25 @@ func validateName(name, typ string) error {
 	return nil
 }
 
+func (desc *IndexDescriptor) allocateName(tableDesc *TableDescriptor, names map[string]struct{}) {
+	segments := make([]string, 0, len(desc.ColumnIDs)+2)
+	segments = append(segments, tableDesc.Name)
+	for _, columnName := range desc.ColumnNames {
+		segments = append(segments, columnName)
+	}
+	segments = append(segments, "idx")
+	baseName := strings.Join(segments, "_")
```

Comment:
Are you following any existing convention here, or is this your own invention. Looks fine either way, though you might want to mention in a comment if this drew inspiration from somewhere.